pkg/ottl/ottlfuncs/README.md

```diff
@@ -353,10 +356,23 @@ The `replace_pattern` function allows replacing all string sections that match a
 
 If one or more sections of `target` match `regex` they will get replaced with `replacement`.
 
+The `replacement` string can refer to matched groups using [regexp.Expand syntax](https://pkg.go.dev/regexp#Regexp.Expand). 
+
 Examples:
 
 - `replace_pattern(resource.attributes["process.command_line"], "password\\=[^\\s]*(\\s?)", "password=***")`
+- `replace_pattern(name, "^kube_([0-9A-Za-z]+_)", "k8s.$1.")` 
 
+Note that when using OTTL within the collector's configuration file, `$` may need to be escaped to `$$` to bypass
```

Comment:
The `$` will always need escaped in the collector, right?  If so, I think the documentation and example should reflect that.  We want the examples to be copy-paste-able.

Reply:
I thought of that, but can't OTTL be used anywhere in the platform, not only config files?  It seems a bit odd to be including escaping at this example when the example is just a generic OTTL strung and not example YAML that would be in a config file.

It's also not clear to me why `\` is escaped in the examples.

Reply:
Since OTTL is a library it can be used in any component and all it cares about is strings so you're correct that `$` is all thats needed in non-config scenarios.  But at the moment 100% of OTTL's usage in Contrib gets the strings from the config, so it is relevant.

Since this line is explicitly calling out the use in configuration we can be strict an say `must be escaped with $$`.  We can add a second sentance that describes this is only needed becuase of how the collector interprets $ in config and that without that restriction you can use $.

> It's also not clear to me why \ is escaped in the examples.
I believe the strings need to escape the backslashes so that are interpreted correctly by the regex compiler.  If thats wrong we should update those examples.

Reply:
> I believe the strings need to escape the backslashes so that are interpreted correctly by the regex compiler.

I just tested it and you are correct.  It's a bit counter-intuitive because `\\` escapes are usually only needed in string literals within compiled code, not necessarily string literals in other contexts (e.g. YAML or JSON files).  It looks like the OTTL lexer intentionally supports escape codes (see [here](https://github.com/alecthomas/participle/issues/130#issuecomment-757465789) for more info).